### PR TITLE
Ft mongo setup

### PR DESCRIPTION
**Pull request by amos-o**
#### What does this PR do?
This PR adds the capability to save links collected to the DB.

#### Description of Task to be completed?
Setup Mongo

#### How should this be manually tested?
1. Make sure to run `npm i` to install new dependencies.
 
2. Update `.env` and add a new key named `MONGODB_REMOTE_URL`

3. Run the bot: `npm start`

4. Post messages with and without links. Refresh the `Links` collection in the db to see your links being saved. Watch the console for logger output.

#### Screenshots (if appropriate)
Shot of the first few links in the db. Notice the `urls` key is an array because some messages can have more than one link.
screen shot 2017-01-17 at 12 59 08 pm

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
+const mongoose = require('mongoose');
 
-const saveToDb = (linksObj) => {
-     if (linksObj.length) {
-        // TODO save to MongoDb
-        return true;
-     }
-    return false;
-}
+const dbUrl = process.env.MONGODB_REMOTE_URL;
```
Comment (JackWachira) [inline lib/utils.js:5]: Fix indent and white lines

File: models/link.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const linkSchema = new Schema({
+  urls: Array, // the extracted URLs
+  text: String, // the whole message
+  user: String, // the user who posted the message
+  channel: String, // the channel it wacleas posted to
+  timestamp: Date // the timestamp of the posting
```
Comment (JackWachira) [inline models/link.js:9]: Is it possible to add the organisationId?

**Review by JackWachira — COMMENTED** [review]
IMO

File: lib/utils.js (hunk 1 above)
Comment (amos-o) [reply lib/utils.js:5]: Ok.

File: models/link.js (hunk 2 above)
Comment (amos-o) [reply models/link.js:9]: The message doesn't come with OrgId. I'm not sure where we can get it from.

**Review by Habu-Kagumba — APPROVED** [review]
Looks good, awesome find -> linkify

**Pull request merged**